packages/edit-site/src/components/secondary-sidebar/inserter-sidebar.js

```diff
@@ -23,23 +25,36 @@ export default function InserterSidebar() {
 	);
 
 	const isMobile = useViewportMatch( 'medium', '<' );
+	const TagName = ! isMobile ? VisuallyHidden : 'div';
 	const [ inserterDialogRef, inserterDialogProps ] = useDialog( {
 		onClose: () => setIsInserterOpened( false ),
+		focusOnMount: null,
 	} );
 
+	const inserterContentRef = useRef();
+	useEffect( () => {
+		inserterContentRef.current
+			.querySelector( '.block-editor-inserter__search input' )
```

Comment:
Could we use pass a `searchInputRef` to `<Library>` and all the way down to the actual `<input>` element here? So that we can omit the query selector, which feels like an implementation detail.

Or we can use `useImperativeHandle` to allow calling `focusSearchInput()` on the `<Library>` ref.

If all of the above can't be done, then I'd prefer using `data-*` attribute over class names for explicitness.

Reply:
@kevin940726 https://github.com/WordPress/gutenberg/pull/37357#discussion_r771228448

I think @youknowriad wanted to avoid doing anything with the block-editor/inserter component because it could be used as a single element. E.g. keep all the implementation in the sidebar files and out of what could be a reusable stand-alone component.

I actually used forwardRef() all the way to the Search component a few commits back.

Reply:
Yep I noticed that, but I think my approach is different because it's not using `forwardRef` to bind the ref to the inserter instance, but passing a different prop `searchInputRef` down to the search input element. `useImperativeHandle` can also work with `forwardRef` so that the ref is referencing the inserter instance but also providing `focusSearchInput()` functionality as well.

Reply:
@kevin940726 I think a ref with an added property is a sane approach indeed. I'm fine either way but we should be mindful of added complexity.